src/watertap_contrib/reflo/unit_models/crystallizer_effect.py

```diff
@@ -0,0 +1,430 @@
+            doc="Steam pressure (gauge) for crystallizer heating: 3 bar default based on Dutta example",
+        )
+
+        self.efficiency_pump = Param(
+            initialize=0.7,
+            mutable=True,
+            units=pyunits.dimensionless,
+            doc="Crystallizer pump efficiency - assumed",
+        )
+
+        self.pump_head_height = Param(
+            initialize=1,
+            mutable=True,
+            units=pyunits.m,
+            doc="Crystallizer pump head height -  assumed, unvalidated",
+        )
+
+        self.energy_flow_superheated_vapor = Var(
+            initialize=1e5,
+            bounds=(-5e6, 5e6),
+            units=pyunits.kilowatt,
+            doc="Energy that could be supplied from vapor",
+        )
+
+        self.delta_temperature_in = Var(
+            self.flowsheet().time,
+            initialize=35,
+            bounds=(None, None),
+            units=pyunits.degK,
+            doc="Temperature difference at the inlet side",
+        )
+
+        self.delta_temperature_out = Var(
+            self.flowsheet().time,
+            initialize=35,
+            bounds=(None, None),
+            units=pyunits.degK,
+            doc="Temperature difference at the outlet side",
+        )
+
+        delta_temperature_chen_callback(self)
+
+        self.heat_exchanger_area = Var(
+            initialize=1000.0,
+            bounds=(0, None),
+            units=pyunits.m**2,
+            doc="Heat exchanger area",
+        )
+
+        self.overall_heat_transfer_coefficient = Var(
+            initialize=0.1,
+            bounds=(0, None),
+            units=pyunits.kilowatt / pyunits.m**2 / pyunits.degK,
+            doc="Overall heat transfer coefficient for heat exchangers",
+        )
+
+        @self.Constraint()
+        def eq_pure_vapor_flow_rate(b):
+            return (
+                b.properties_pure_water[0].flow_mass_phase_comp["Vap", "H2O"]
+                == b.properties_in[0].flow_mass_phase_comp["Vap", "H2O"]
+            )
+
+        @self.Constraint(doc="Thermal energy in the vapor")
+        def eq_vapor_energy_constraint(b):
+            return b.energy_flow_superheated_vapor == (
+                b.properties_vapor[0].flow_mass_phase_comp["Vap", "H2O"]
+                * (
+                    b.properties_pure_water[
+                        0
+                    ].dh_vap_mass_solvent  # Latent heat from the vapor
+                    + b.properties_vapor[0].enth_mass_solvent["Vap"]
+                    - b.properties_pure_water[0].enth_mass_solvent["Vap"]
+                )
+            )
+
+        self.del_component(self.eq_p_con1)
+        self.del_component(self.eq_p_con2)
+
+        @self.Constraint()
+        def eq_p_con1(b):
+            return b.pressure_operating == b.properties_out[0].pressure
+
+        @self.Constraint()
+        def eq_p_con2(b):
+            return b.pressure_operating == b.properties_solids[0].pressure
+
+        @self.Constraint()
```

Comment:
Is there an `eq_p_con3` this is still being used from the parent class?

Reply:
Yes. Parent class has ```eq_q_con1```, ```eq_q_con2```, and ```eq_q_con3```.